src/App.js

```diff
@@ -0,0 +1,27 @@
+import { Route, Routes } from 'react-router-dom';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import Main from './components/Main';
+import Home from './components/Home';
+import Detail from './components/Detail';
+import { fetchCoutries } from './features/country/countrySlice';
+import './index.css';
+
+function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCoutries());
+  });
+
+  return (
+    <Routes>
+      <Route path="/" element={<Main />}>
+        <Route index element={<Home />} />
+        <Route path="/details" element={<Detail />} />
```

Comment:
- [ Required ] As suggested by the previous reviewer I think it would be better if the pattern of the details page URL follows this pattern `/details/:id`. This will ensure that all the items on your web page have a Unique details URL for example the URL that navigates to Kenya could be `http://localhost:3000/details/kenya` and the URL that navigates to Tunisia could be `http://localhost:3000/details/tunisia`. Please see the pattern below and rectify this.

## Change line 21 on App.js to this
```JavaScript
<Route exact path="/details/:id" element={<Detail />}  />
```
## Change line 20-24 in Home component to this
```JavaScript
  <Link
       className={((index + 2) % 2) === 0 ? 'country' : 'country  pair-red'}
       to={`/details/${item.name.common}`}
       state={{ country: item.name.common }}
   >
```